### DISCUSSION TITLE
Wildcards in Distiguished Name for authentication

### DISCUSSION THREAD

**Discussion by amdrsantos** [strongswan/strongswan#1074] [Q&A]
According to [Matching CN field in Distinguished Names](https://docs.strongswan.org/docs/5.9/support/faq.html#_matching_cn_field_in_distinguished_names), we can not set the connections.<conn>.<local|remote>.id field to match a specific CN in an X509 certificate.

But in https://www.strongswan.org/docs/readme4.htm#section_4.6, this is still possible.

From what I checked, it works if the id string contains the same number of RDNs. For example:
strongSwan can match the id "C=*, O=*, OU=*, CN=testName" with certificate DN "C=CH, O=ACME, OU=Research, CN=testName"

For local authentication, this seems to be working fine. But for remote authentication, I end up with a half-open IKE, which then is destroyed.

Do you have some idea, of what is happening?

**Comment by tobiasbrunner** [comment]
> According to [Matching CN field in Distinguished Names](https://docs.strongswan.org/docs/5.9/support/faq.html#_matching_cn_field_in_distinguished_names), we can not set the connections..<local|remote>.id field to match a specific CN in an X509 certificate.

The FAQ entry might be a bit imprecise. You can obviously match CNs when comparing DNs, but you can't match other identities against it or parts of it. For instance, if you have a certificate with `..., CN=vpn.strongswan.org` then the FQDN identity `vpn.strongswan.org` won't match that certificate, it has to contain a SAN of type dNSName with that FQDN (unlike e.g. with TLS where many clients match domain names against CNs of server certificates).

> From what I checked, it works if the id string contains the same number of RDNs.

This could even be simplified with the `charon.rdn_matching` option. Setting it to `relaxed` allows to match the configured remote identity `CN=testName` against your example certificate because all missing RDNs are assumed to be wildcards.

> But for remote authentication, I end up with a half-open IKE, which then is destroyed.

What do you mean? Any configs/logs?
